src/main/java/org/jenkinsci/plugins/mesos/MesosComputer.java

```diff
@@ -43,4 +50,10 @@ public HttpResponse doDoDelete() throws IOException {
       getNode().terminate();
     return new HttpRedirect("..");
   }
+
+  @Override
+  @WebMethod(name="slave-agent.jnlp")
+  public HttpResponse doSlaveAgentJnlp(StaplerRequest req, StaplerResponse res) throws IOException, ServletException {
```

Comment:
To override the default slave-agent.jnlp.jelly with the new file mesos-slave-agent.jnlp.jelly which would enable slave.jar to talk to only the Jenkins URL configured in Mesos plugin configuration.